universal-application-tool-0.0.1/app/repository/QuestionRepository.java

```diff
@@ -33,6 +44,56 @@ public QuestionRepository(EbeanConfig ebeanConfig, DatabaseExecutionContext exec
         () -> ebeanServer.find(Question.class).setId(id).findOneOrEmpty(), executionContext);
   }
 
+  public Question getOrCreateDraft(QuestionDefinition definition) {
+    try {
+      Transaction transaction = ebeanServer.beginTransaction();
+      Optional<Question> draftMaybe =
+          ebeanServer
+              .find(Question.class)
+              .usingTransaction(transaction)
+              .where()
+              .eq("name", definition.getName())
+              .eq("lifecycle_stage", LifecycleStage.DRAFT.getValue())
+              .findOneOrEmpty();
+      if (draftMaybe.isPresent()) {
+        return draftMaybe.get();
+      }
+      int existingQuestionCount =
+          ebeanServer
+              .find(Question.class)
+              .usingTransaction(transaction)
+              .where()
+              .eq("name", definition.getName())
+              .findCount();
+      Question newDraft =
+          new Question(
+              new QuestionDefinitionBuilder(definition)
+                  .setId(null)
+                  .setVersion(existingQuestionCount + 1L)
+                  .build());
+      newDraft.setLifecycleStage(LifecycleStage.DRAFT);
+      newDraft.id = null;
```

Comment:
Question (blocking): why is setting id to null here necessary?

Reply:
Oooh I was confused because I mistook line 81 for setting the ID on the question.